// @storex/utils/lib/async/delay-state.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class DelayState {
    constructor(target, opt, theThisFunction) {
        this.target = target;
        this.opt = opt;
        this.theThisFunction = theThisFunction;
        this.state = {
            waiting: false,
            delayCallingAmount: 0,
            callingAmount: 0,
            argsState: undefined,
        };
        this._clearState = () => {
            this.state.waiting = false;
            this.state.callingAmount = 0;
            this.state.delayCallingAmount = 0;
        };
        this.delay = () => {
            setTimeout(() => {
                this._clearState();
                this.invoke();
            }, this.opt.time);
        };
        this.invoke = () => {
            this.target.apply(this.theThisFunction, [this.state.argsState]);
            // clearing the args state
            this.state.argsState = undefined;
            //this.delay();
        };
        if (opt.startDelayAfter == undefined) {
            opt.startDelayAfter = 1;
        }
    }
}
exports.DelayState = DelayState;
